#! /bin/sh /usr/share/dpatch/dpatch-run
## 03_allow_unsafe_chars_in_password.dpatch by Javier Uruen Val <javi@warp.es>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Backport 0.11.100 to allow unsafe characters in user passwords

@DPATCH@

Index: usersandgroups/src/EBox/CGI/ModifyUser.pm
===================================================================
--- usersandgroups/src/EBox/CGI/ModifyUser.pm	(revisión: 9763)
+++ usersandgroups/src/EBox/CGI/ModifyUser.pm	(copia de trabajo)
@@ -57,9 +57,9 @@
 			 };
 	
 	# Change password if not empty		 
-	my $password = $self->param('password');
+	my $password = $self->unsafeParam('password');
 	if ($password) {
-		my $repassword = $self->param('repassword');
+		my $repassword = $self->unsafeParam('repassword');
 		if ($password ne $repassword){
 			 throw EBox::Exceptions::External(
 					__('Passwords do not match.'));
Index: usersandgroups/src/EBox/CGI/AddUser.pm
===================================================================
--- usersandgroups/src/EBox/CGI/AddUser.pm	(revisión: 9763)
+++ usersandgroups/src/EBox/CGI/AddUser.pm	(copia de trabajo)
@@ -51,8 +51,8 @@
 	my $user;
 	$user->{'user'} = $self->param('username');
 	$user->{'fullname'} = $self->param('fullname');
-	$user->{'password'} = $self->param('password');
-	$user->{'repassword'} = $self->param('repassword');
+	$user->{'password'} = $self->unsafeParam('password');
+	$user->{'repassword'} = $self->unsafeParam('repassword');
 	$user->{'group'} = $self->param('group');
 	$user->{'comment'} = $self->param('comment');
 
